### PR TITLE
spring-boot-1

### PR DESCRIPTION
**Pull request by ShemsheyMaxim**

### CODE REVIEW

File: src/main/java/com/springboot/onlinebanking/repository/RoleRepository.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.springboot.onlinebanking.repository;
+
+import com.springboot.onlinebanking.model.Role;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+public interface RoleRepository extends JpaRepository<Role, Long> {
+    @Query("SELECT r FROM Role r WHERE r.roleName = :roleName")
+    Optional<Role> getRoleByRoleName(String roleName);
```
Comment (Sashk52) [inline src/main/java/com/springboot/onlinebanking/repository/RoleRepository.java]: ....getRoleByRoleName( @Param("roleName")String roleName));

I think you have add @Param here.

**Review by Sashk52 — CHANGES_REQUESTED** [review]

**Review by Sashk52 — APPROVED** [review]

File: src/main/java/com/springboot/onlinebanking/repository/RoleRepository.java (hunk 1 above)
Comment (boroda4436) [reply src/main/java/com/springboot/onlinebanking/repository/RoleRepository.java]: @Sashk52 `@Param` is not required here

File: src/main/java/com/springboot/onlinebanking/repository/RoleRepository.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.springboot.onlinebanking.repository;
+
+import com.springboot.onlinebanking.model.Role;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+
+public interface RoleRepository extends JpaRepository<Role, Long> {
+    @Query("SELECT r FROM Role r WHERE r.roleName = :roleName")
```
Comment (boroda4436) [inline src/main/java/com/springboot/onlinebanking/repository/RoleRepository.java]: @ShemsheyMaxim Can we avoid using `@Query` here?

File: src/main/java/com/springboot/onlinebanking/repository/UserRepository.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.springboot.onlinebanking.repository;
+
+import com.springboot.onlinebanking.model.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface UserRepository extends JpaRepository<User, Long> {
```
Comment (boroda4436) [inline src/main/java/com/springboot/onlinebanking/repository/UserRepository.java:7]: @ShemsheyMaxim Do we need to add `findByNumber`?

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/com/springboot/onlinebanking/repository/RoleRepository.java (hunk 2 above)
Comment (ShemsheyMaxim) [reply src/main/java/com/springboot/onlinebanking/repository/RoleRepository.java]: done

File: src/main/java/com/springboot/onlinebanking/repository/UserRepository.java (hunk 3 above)
Comment (ShemsheyMaxim) [reply src/main/java/com/springboot/onlinebanking/repository/UserRepository.java:7]: done

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**